arelle/api/SessionResults.py

```diff
@@ -0,0 +1,35 @@
+"""
+See COPYRIGHT.md for copyright information.
+"""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from arelle.Cntlr import Cntlr
+from arelle.ModelXbrl import ModelXbrl
+
+
+@dataclass
+class SessionResults:
+    _cntlr: Cntlr
+
+    def __enter__(self) -> Any:
+        return self
+
+    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
+        self.close()
+
+    def get_logs(self) -> str:
+        handler = self._cntlr.logHandler
+        if hasattr(handler, 'getXml'):
+            return str(handler.getXml(clearLogBuffer=False, includeDeclaration=False))
+        if hasattr(handler, 'getText'):
+            return str(handler.getText(clearLogBuffer=False))
+        return ""
+
+    def get_models(self) -> list[ModelXbrl]:
+        return self._cntlr.modelManager.loadedModelXbrls
+
+    def close(self) -> None:
+        self._cntlr.close()
```

Comment:
SessionResults closing the Cntlr seems problematic. What happens if you call run more than once and have multiple SessionResults objects?